ethereum/contracts/ETHApp.sol

```diff
@@ -93,13 +121,31 @@ contract ETHApp is RewardSource, AccessControl {
         bytes32 _recipient,
         uint256 _amount
     ) private pure returns (bytes memory) {
-        return
-            abi.encodePacked(
+        return abi.encodePacked(
+                MINT_CALL,
+                _sender,
+                bytes1(0x00), // Encode recipient as MultiAddress::Id
+                _recipient,
+                _amount.encode256(),
+                bytes1(0x00)
+            );
+    }
+
+    // SCALE-encode payload with parachain Id
+    function encodeCall(
```

Comment:
To reduce the chance of bugs being introduced in future, rather explicitly give these `encodeCall` functions different names, such as `encodeCall` and `encodeCallWithParaId`, etc.